### PR TITLE
#17901 Fixing error in rules permission

### PR DESCRIPTION
**Pull request by freddyucv**
Using new RuleAPI methid ti get rules by parent ID

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/sites/ruleengine/rules/RuleResource.java [hunk 1]
```diff
@@ -82,12 +77,11 @@ protected RuleResource(ApiProvider apiProvider, WebResource webResource) {
     public Map<String, RestRule> list(
             @Context final HttpServletRequest request,
             @Context final HttpServletResponse response,
-            @PathParam("siteId") String siteId) throws DotSecurityException, DotDataException {
+            @PathParam("siteId") String parentId) throws DotSecurityException, DotDataException {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/sites/ruleengine/rules/RuleResource.java]: Codacy Issue found: [The String literal "siteId" appears 5 times in this file; the first occurrence is on line 80](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5156545)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 2]
```diff
@@ -1,15 +1,15 @@
 package com.dotcms.enterprise.rules;
 
-import com.dotcms.datagen.RoleDataGen;
-import com.dotcms.datagen.SiteDataGen;
-import com.dotcms.datagen.UserDataGen;
+import com.dotcms.datagen.*;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotcms.datagen'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5156545)

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/sites/ruleengine/rules/RuleResource.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/sites/ruleengine/rules/RuleResource.java]: Codacy Issue found: [Avoid reassigning parameters such as 'parentId'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5156545)

Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/sites/ruleengine/rules/RuleResource.java]: Codacy Issue found: [The String literal "siteId" appears 5 times in this file; the first occurrence is on line 80](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5156545)

Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/sites/ruleengine/rules/RuleResource.java]: Codacy Issue found: [Avoid reassigning parameters such as 'parentId'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5156545)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotcms.datagen'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5156545)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 3]
```diff
@@ -51,6 +53,63 @@ public static void prepare() throws Exception {
...
+
+        try {
+            rulesAPI.deleteRule(rule, user, false);
+            throw new AssertionError("DotSecurityException Expected");
+        } catch (DotSecurityException e) {
+            assertFalse(existRule(user, rule));
+        }
+    }
+
+    private boolean existRule(User user, Rule rule) throws DotDataException, DotSecurityException {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [Parameter 'user' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5156545)

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/sites/ruleengine/rules/RuleResource.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/sites/ruleengine/rules/RuleResource.java]: Codacy Issue found: [The String literal "siteId" appears 5 times in this file; the first occurrence is on line 80](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5156545)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/rules/RuleDataGen.java [hunk 4]
```diff
@@ -1,17 +1,20 @@
 package com.dotmarketing.portlets.rules;
 
+import com.dotcms.datagen.AbstractDataGen;
 import com.dotmarketing.beans.Host;
 import com.dotmarketing.business.APILocator;
 import com.dotmarketing.portlets.contentlet.business.HostAPI;
 import com.dotcms.enterprise.rules.RulesAPI;
 //import com.dotmarketing.portlets.rules.conditionlet.MockTrueConditionlet;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/rules/RuleDataGen.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.portlets.contentlet.model.Contentlet'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5156545)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotcms.datagen'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5156545)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 5]
```diff
@@ -27,10 +28,9 @@
 import java.util.List;
 import java.util.stream.Collectors;
 
+import static com.dotcms.util.CollectionsUtils.list;
 import static org.jgroups.util.Util.assertFalse;
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNull;
-import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.*;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [Avoid unused imports such as 'org.junit.Assert'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5156545)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 6]
```diff
@@ -51,6 +51,178 @@ public static void prepare() throws Exception {
...
+
+        try {
+            rulesAPI.deleteRule(rule, user, false);
+            throw new AssertionError("DotSecurityException Expected");
+        } catch (DotSecurityException e) {
+            assertFalse(existRule(user, rule));
+        }
+    }
+
+    private boolean existRule(User user, Rule rule) throws DotDataException, DotSecurityException {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [Parameter 'user' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5156545)

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/sites/ruleengine/rules/RuleResource.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/sites/ruleengine/rules/RuleResource.java]: Codacy Issue found: [Avoid reassigning parameters such as 'parentId'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5156545)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 7]
```diff
@@ -81,6 +253,95 @@ public void shouldGetRules() throws DotSecurityException, DotDataException {
...
+        final HTMLPageAsset anotherHtmlPageAsset = new HTMLPageDataGen(host, template).nextPersisted();
+        final Rule rule3 = new RuleDataGen().page(anotherHtmlPageAsset).nextPersisted();
+
+        addPermissionToReadRulesFolder(role, folder);
+
+        final List<Rule> rules = rulesAPI.getAllRulesByParent(htmlPageAsset.getIdentifier(), user, false);
+
+        assertEquals(2, rules.size());
+
+        final List<String> rulesId = rules.stream().map(rule -> rule.getId()).collect(Collectors.toList());
```
Comment (nollymar) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: why are you comparing against 3 rules if the assert below verifies that there are just 2 rules in the collection?

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 8]
```diff
@@ -119,6 +418,30 @@ public void shouldReturnEmptyList() throws DotSecurityException, DotDataExceptio
...
+        final Host host = new SiteDataGen().nextPersisted();
+        final Template template = new TemplateDataGen().nextPersisted();
+        final HTMLPageAsset htmlPageAsset = new HTMLPageDataGen(host, template).nextPersisted();
+        final Rule rule = new RuleDataGen().page(htmlPageAsset).next();
+
+        this.addPermission(role, host, true);
+        rulesAPI.saveRule(rule, user, false);
+
+        final List<Rule> rules = rulesAPI.getAllRulesByParent(htmlPageAsset, systemUser, true);
+        assertEquals(1, rules.size());
```
Comment (nollymar) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: I think you should also compare the parent, not just the size

**Review by nollymar — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 9]
```diff
@@ -51,6 +51,178 @@ public static void prepare() throws Exception {
...
+
+        List<Rule> allRules = rulesAPI.getAllRules(user, false);
+        assertEquals(1, allRules.size());
+
+        final Rule ruleFromDataBase = allRules.get(0);
+
+        final List<ConditionGroup> groups = ruleFromDataBase.getGroups();
+        assertEquals(1, groups.size());
+
+        assertEquals(groups.get(0).getOperator(), conditionGroup.getOperator());
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: I think the expected value goes at first

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 10]
```diff
@@ -51,6 +51,178 @@ public static void prepare() throws Exception {
...
+
+        //Saving and testing Condition
+        final Condition condition = new ConditionDataGen().group(groups.get(0)).next();
+
+        rulesAPI.saveCondition(condition, user, false);
+
+        allRules = rulesAPI.getAllRules(user, false);
+        assertEquals(1, allRules.size());
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: this assert doesn't look like it's needed

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 11]
```diff
@@ -51,6 +51,178 @@ public static void prepare() throws Exception {
...
+        final Rule rule = new RuleDataGen().host(host).nextPersisted();
+
+        this.addPermission(role, host, true);
+
+        final RuleAction ruleAction = new RuleActionDataGen().rule(rule).next();
+
+        rulesAPI.saveRuleAction(ruleAction, user, false);
+
+        final List<Rule> allRules = rulesAPI.getAllRules(user, false);
+        assertEquals(1, allRules.size());
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: assert not needed

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java (hunk 10 above)
Comment (freddyucv) [reply dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: done https://github.com/dotCMS/core/pull/18127/commits/d8a32ec5a775f434c7ee34caeaa3b0e19018ab5f#diff-9baded4fb0fb63e2648d9c9c4380ec1aL93

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java (hunk 11 above)
Comment (freddyucv) [reply dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: done https://github.com/dotCMS/core/pull/18127/commits/d8a32ec5a775f434c7ee34caeaa3b0e19018ab5f#diff-9baded4fb0fb63e2648d9c9c4380ec1aL142

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/sites/ruleengine/rules/RuleResource.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/sites/ruleengine/rules/RuleResource.java]: Codacy Issue found: [The String literal "siteId" appears 5 times in this file; the first occurrence is on line 80](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5156545)

Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/sites/ruleengine/rules/RuleResource.java]: Codacy Issue found: [Avoid reassigning parameters such as 'parentId'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5156545)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 12]
```diff
@@ -1,15 +1,14 @@
 package com.dotcms.enterprise.rules;
 
-import com.dotcms.datagen.RoleDataGen;
-import com.dotcms.datagen.SiteDataGen;
-import com.dotcms.datagen.UserDataGen;
-import com.dotcms.util.CollectionsUtils;
+import com.dotcms.datagen.*;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotcms.datagen'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5156545)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java [hunk 13]
```diff
@@ -27,10 +27,9 @@
 import java.util.List;
 import java.util.stream.Collectors;
 
+import static com.dotcms.util.CollectionsUtils.list;
 import static org.jgroups.util.Util.assertFalse;
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNull;
-import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.*;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java]: Codacy Issue found: [Avoid unused imports such as 'org.junit.Assert'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5156545)

**Review by dsilvam — APPROVED** [review]

**Pull request merged**